### PR TITLE
integrated airbrake into middleware

### PR DESCRIPTION
**Pull request by malekascha**
Note: after this change goes live, we'll need to edit the router creation in atlas backend and admin backend to pass in the necessary airbrake info.

### CODE REVIEW

**Review by MSamman — CHANGES_REQUESTED** [review]
I think there is a better way to do this. Instead of changing the signature of `NewRouter()`, I think you should add a function the the router struct that looks something like the following:

`(g *Router) EnableAirbrake(airbrakeId, airbrakeProject string)`

That way, we don't break any existing NewRouter calls when we updated easyroute in our projects and we can still easily enable airbrake by calling this one new function.

File: router.go [hunk 1]
```diff
@@ -68,13 +89,21 @@ func (g *Router) SubRouteC(prefix string, beforeFn beforeHandlerFunc) Router {
...
+		"",
+		false,
 	}
 
 	return router
 }
 
 func (g *Router) requestHandler(fn handlerFunc) http.HandlerFunc {
 	return func(w http.ResponseWriter, r *http.Request) {
+		if(g.airbrakeEnabled == true){
```
Comment (leehsueh) [inline router.go]: My normal style OCD comments apply here:

`if g.airbrakeEnabled {...`

File: router.go [hunk 2]
```diff
@@ -22,17 +24,33 @@ type Router struct {
...
 }
 
 // NewRouter creates a new easyroute Router object with the provided
 // before handler and logger struct
-func NewRouter(beforeFn beforeHandlerFunc, logger Logger) Router {
+func NewRouter(beforeFn beforeHandlerFunc, logger Logger, airbrakeInfo map[string]string) Router {
 	muxRouter := mux.NewRouter()
-
+	var enableAirbrake bool
+	if(airbrakeInfo["enable"] == "true"){
```
Comment (leehsueh) [inline router.go]: Same style comment here. Remove parenthesis.

File: router.go [hunk 3]
```diff
@@ -22,17 +24,33 @@ type Router struct {
...
+		enableAirbrake = true
+	} else {
+		enableAirbrake = false
+	}
+	airbrakeId, err := strconv.Atoi(airbrakeInfo["projectId"])
+	if(err != nil){
```
Comment (leehsueh) [inline router.go]: And here. Remove parentheses

**Comment by malekascha** [comment]
Excellent points, @MSamman @leehsueh. I've made changes to reflect them

**Pull request merged**